internal/logs/traces/table.go

```diff
@@ -0,0 +1,777 @@
+package traces
+
+import (
+	"time"
+
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/meta"
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/xerrors"
+	"github.com/ydb-platform/ydb-go-sdk/v3/logs"
+	"github.com/ydb-platform/ydb-go-sdk/v3/retry"
+	"github.com/ydb-platform/ydb-go-sdk/v3/trace"
+)
+
+// Table makes trace.Table with logging events from details
+//
+//nolint:gocyclo
+func Table(l logs.Logger, details trace.Details, opts ...Option) (t trace.Table) {
+	if details&trace.TableEvents == 0 {
+		return
+	}
+	options := ParseOptions(opts...)
+	ll := newLogger(l, "table")
+	t.OnDo = func(
+		info trace.TableDoStartInfo,
+	) func(
+		info trace.TableDoIntermediateInfo,
+	) func(
+		trace.TableDoDoneInfo,
+	) {
+		idempotent := info.Idempotent
+		ll.Trace(`do start`,
+			logs.Bool("idempotent", idempotent),
+		)
+		start := time.Now()
+		return func(info trace.TableDoIntermediateInfo) func(trace.TableDoDoneInfo) {
+			if info.Error == nil {
+				ll.Trace(`do attempt done`,
+					logs.Duration("latency", time.Since(start)),
+					logs.Bool("idempotent", idempotent),
+				)
+			} else {
+				f := ll.Warn
```

Comment:
What about direct use log method with level as argument instead of dinamic select proxy-function?